lib/ansible/modules/source_control/git.py

```diff
@@ -432,6 +439,21 @@ def clone(git_path, module, repo, dest, remote, depth, version, bare,
                         "HEAD, branches, tags or in combination with refspec.")
     if reference:
         cmd.extend(['--reference', str(reference)])
+    separate_git_dir_fallback = False
+    if separate_git_dir is not None and bare:
+        module.warn("Ignoring separate_git_dir argument. "
+                    "Can not do bare clone with argument separate_git_dir.")
+    elif separate_git_dir:
+        git_version_used = git_version(git_path, module)
+        if git_version_used is None:
+            raise AnsibleError("Can not find git executable.")
+        if git_version_used < LooseVersion('1.7.5'):
```

Comment:
It might be a better idea to feature-detect its support using smth like `git clone --help | grep '\-\-separate-git-dir'` (while a bit more expensive).

Reply:
ok, I'm now fixing fatal bugs and after the module runs well I will implement this.